tests/test_doctestplus.py

```diff
@@ -690,3 +690,35 @@ def f():
     reprec.assertoutcome(passed=1)
     assert reprec.listoutcomes()[0][0].location[0] == os.path.join('test', 'a', 'testcode.py')
     assert caplog.text == ''
+
+
+def test_remote_data(testdir):
+    testdir.makeini(
+        """
+        [pytest]
+        doctestplus = enabled
+    """)
+
+    # should be ignored
+    p = testdir.makefile(
+        '.rst',
+        """
+        .. doctest-remote-data::
+            >>> 1 + 1
+            2
+        """
+    )
```

Comment:
```suggestion
    p = testdir.makefile(
        '.rst',
        """                                                                                                             
                                                                                                                        
        # This one should fail, but we skip it, thus the test should pass.                                              
        .. doctest-remote-data::                                                                                        
            # Now do real remote access:                                                                                
            >>> from contextlib import closing                                                                          
            >>> from urllib.request import urlopen                                                                      
            >>> with closing(urlopen('https://www.astropy.org')) as remote:                                             
            ...     remote.read()    # doctest: +IGNORE_OUTPUT                                                          
            #                                                                                                           
            # And do some failing to check whether the whole block is indeed skipped                                    
            >>> 1 + 1                                                                                                   
            3                                                                                                           
                                                                                                                        
        """
```

Reply:
this should generate a skipped status, and a failed status depending on whether the `remote-data` is used or not.

Unfortunately, the one whole doc file is part of one big test, I suppose that is the reason you couldn't generate more outcome (it maybe related to https://github.com/astropy/pytest-doctestplus/issues/44)